flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala

```diff
@@ -215,6 +218,74 @@ abstract class TableEnvironment(val config: TableConfig) {
     rels.map(_.asInstanceOf[ExecNode[_, _]])
   }
 
+  /**
+    * Register an [[ReadableCatalog]] under a unique name.
+    *
+    * @param name the name under which the catalog will be registered
+    * @param catalog the catalog to register
+    * @throws CatalogAlreadyExistsException thrown if the catalog already exists
+    */
+  @throws[CatalogAlreadyExistsException]
+  def registerCatalog(name: String, catalog: ReadableCatalog): Unit = {
+    catalogManager.registerCatalog(name, catalog)
+  }
+
+  /**
+    * Get a registered [[ReadableCatalog]].
+    *
+    * @param catalogName name of the catalog to get
+    * @return the requested catalog
+    * @throws CatalogNotExistException thrown if the catalog doesn't exist
+    */
+  @throws[CatalogNotExistException]
+  def getCatalog(catalogName: String): ReadableCatalog = {
+    catalogManager.getCatalog(catalogName)
+  }
+
+  /**
+    * Get the current catalog.
+    *
+    * @return the current catalog in CatalogManager
+    */
+  def getCurrentCatalog(): ReadableCatalog = {
+    catalogManager.getCurrentCatalog
+  }
+
+  /**
+    * Get the current database name.
+    *
+    * @return the current database of the current catalog
+    */
+  def getCurrentDatabaseName(): String = {
+    catalogManager.getCurrentCatalog.getCurrentDatabase
+  }
+
+  /**
+    * Set the current catalog.
+    *
+    * @param name name of the catalog to set as current catalog
+    * @throws CatalogNotExistException thrown if the catalog doesn't exist
+    */
+  @throws[CatalogNotExistException]
+  def setCurrentCatalog(name: String): Unit = {
+    catalogManager.setCurrentCatalog(name)
+  }
+
+  /**
+    * Set the current catalog and current database.
+    *
+    * @param catalogName name of the catalog to set as current catalog
+    * @param databaseName name of the database to set as current database
+    * @throws CatalogNotExistException  thrown if the catalog doesn't exist
+    * @throws DatabaseNotExistException thrown if the database doesn't exist
+    */
+  @throws[CatalogNotExistException]
+  @throws[DatabaseNotExistException]
+  def setCurrentDatabase(catalogName: String, databaseName: String): Unit = {
+    catalogManager.setCurrentCatalog(catalogName)
+    catalogManager.getCurrentCatalog.setCurrentDatabase(databaseName)
```

Comment:
"default database" is only a default value of "current database", and yes, "current database" is a session property.

I'm not strong on where to store the "current database". I prefer this way because 1) since every catalog can have such a property, we would need a map in CatalogManager to maintain the mapping then 2) it matches well with [catalog configs in SQL Client yaml file ](https://docs.google.com/document/d/1ALxfiGZBaZ8KUNJtoT443hReoPoJEdt9Db2wiJ2LuWA/edit?usp=sharing)

Reply:
There can only be a single current database per session, not a one current dB per catalog. So we don't need a `Map` in `CatalogManager`.
Have you checked what SQL standard says about it? As I understand it and checked it there is always a default prefix per session (either catalog or catalog + schema - equivalent to DB), so a current DB is not set per catalog.

What if the same catalog is used by multiple users for different sessions? How do you retrieve the original default DB for a new session? As I said originally, this is a property of a session and I strongly believe it should not be stored in the catalog. WDYT?

Reply:
I agree the "current database" thing is now confusing because it actually presents two things: 1) the database of the session if its catalog is current; 2) which will be the current db when users switch to its catalog, it will be the "default_db" in each catalog if users don't overwrite it, that's why CatalogManager needs to keep a map if we wanna extract it out. We separated these two concepts In Blink, and we probably do that too in Flink by replacing `get/setCurrentDatabase` with `String getDefaultDatabase()` (if that's way, we should do it in a different JIRA/PR).

IMHO, this is not important at this moment, as it's only a small usability thing from users' perspective. I can remove parts related to `setCurrentDatabase` from this PR, and we can bake it for more time.